File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.java
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        int sum=0;
        for(int n:apple)
        {
            sum+=n;
        }
        int[] des=new int [capacity.length];
        for(int i=0;i<capacity.length;i++)
        {
            for(int j=i+1;j<capacity.length;j++)
            {
                if(capacity[i]<capacity[j])
                {
                    int temp=capacity[i];
                    capacity[i]=capacity[j];
                    capacity[j]=temp;

                }
            }

        }
        int box=0;
        for(int i=0;i<capacity.length;i++)
        {
            sum-=capacity[i];
            box++;
            if(sum<=0)
            {
                break;
            }

        }
        return box;

    }
}

File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.js
/**
 * @param {number[]} apple
 * @param {number[]} capacity
 * @return {number}
 */
var minimumBoxes = function(apple, capacity) {
    let sum=apple.reduce((a,b)=>(a+b),0);
    capacity.sort((a,b)=>(b-a));
    let box=0;
    for(let i of capacity)
    {
        sum-=i;
        box++;
        if(sum<=0)
        {
            break;
        }
    }
    return box;
};